test/spec/components/alerts.component.spec.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+describe("Component: Alerts", function(){
+
+  var $scope, $compile, element;
+  beforeEach(inject(function(_$compile_, _$rootScope_) {
+      $scope = _$rootScope_;
+      $compile = _$compile_;
+      $scope.alerts = {
+        danger: {
+          type: "error",
+          message: "An error happened and stuff.",
+          links: [
+            {
+              label: "test_link"
+            }
+          ]
+        },
+        warning: {
+          type: "warning",
+          message: "I'm going to count to five."
+        },
+        success: {
+          type: "success",
+          message: "Huzzah!"
+        },
+        info: {
+          message: "This is informational."
+        }
+      };
+
+      // Create an instance of the component
+      element = angular.element('<alerts alerts="alerts"></alerts>');
+      element = $compile(element)($scope); // Compile the component
+      $scope.$digest(); // Update the HTML
+  }));
+
+  // Ensure four alert elements exist
+  it("contain alert elements", function () {
+    let allAlerts = element.find('.alert');
+    expect(allAlerts.length).toBe(4); // four alerts exist
+  });
+
+  // Ensure danger alert exists and is correct
```

Comment:
Generally pretty good with tests.  Do we need to test negative cases along with positive cases?  Example, `warning` should not have `alert-success`, etc.  That could be a bit tedious for this, tho I think its good practice to test both sides.